### DISCUSSION TITLE
Django Commons Organizer/Admin initial dicussion

### DISCUSSION THREAD

**Discussion by tim-schilling** [django-commons/membership#19] [General]
Hi folks! I deeply appreciate the interest all of you have shown. It means a lot to see that type of response. My intention is for this to be a community organization so I'd like for those that are interested in participating to define a self-selection policy. The one caveat is around PyPI access. That needs to be limited to at most three people at any given time (excluding project team admins). But we'll get to that later.

~~**Ahead of our first discussion, [please fill out this form](https://forms.gle/DkMxy8m19ssDosX7A) and [this meeting poll](https://xoyondo.com/dp/wa1thb94i9xw1ju)**~~ This is closed now.

For our initial discussion I'd like us to achieve the following:
- Achieve understanding on what the organization's purpose for the next 6 months
- Identify responsibilities for those 6 months (see form)
- Define meeting cadence and communication preferences
- Identify next tasks for each group/person

To expedite our process a bit, I'm going to start us off with a few agreements. We can revisit them in the future, but it's easier to plan when there are some guidelines to our actions.

1. The tenets of the org are community support and sustainability
2. Prioritize package maintenance over defining policy/best practices
3. Prioritize simplicity and repeatability over customization and edge-cases

cc: @knyghty @joshuadavidthomas @matthiask @benbacardi @carltongibson @williln @Stormheg @ryancheley @marksweb

**Comment by marksweb** [comment]
I feel like a wise early job would be applying for a pypi organisation.

**Reply by tim-schilling** [reply]
Agreed. I sent that in a month ago.

**Reply by tim-schilling** [reply]
Though it was for a different name. Hopefully changing that isn't super hard 😅

**Reply by thibaudcolas** [reply]
For what it’s worth – It’s been 3 months (March 6th) that I requested a PyPI organization for [wagtail-nest](https://github.com/wagtail-nest). No news yet.

**Reply by Stormheg** [reply]
Does anyone here have a contact at the PyPI team who could help speed up the application?

**Reply by tim-schilling** [reply]
On the Django Chat episode with Deb, it sounded like the organization feature of PyPI has technical issues and the feature itself isn't complete. If that's true, then it's not an application backlog, but rather a project delay. I could be misremembering though.

**Comment by carltongibson** [comment]
I filled in the form. I found it quite hard to answer meaningfully, as I'm not sure yet what's needed. Please take middle row answers as 🤷 IDK.

**Reply by marksweb** [reply]
Yeah likewise. 👍

**Reply by tim-schilling** [reply]
That's fair! I wanted to at least try to gather some idea of what people were expecting to do / get out of it.

**Reply by tim-schilling** [reply]
Is there something else I could provide that would help clarify or is it the nature of the situation?

**Reply by carltongibson** [reply]
You clearly have some broader "areas" in mind... maybe bucketing the roles into fewer of those...? 

My (limited) experience of these things is that there can be a danger of having more structure than there is really org to support it. 

Making sure CoC is in place. Responding to new contacts. Managing repos (and systematising that). ... what more is there? – I'm not sure we know yet do we? 

🤔

My inclination would be to flex into all
of that with only the minimum spelled out up front. 

Maybe there are more clear things though?

**Reply by tim-schilling** [reply]
That's generally it with regular communication to confirm people are happy with their roles.

**Reply by williln** [reply]
I answered entirely "maybe" -- sorry for not being more descriptive! I do feel like I need more orientation into what this is, what all these things mean, what the time commitments (including response time -- I am clearly not a prompt responder 😬 ) so I just didn't want to commit, or opt out, of anything without more information.

**Reply by benbacardi** [reply]
Likewise, all my "maybes" were more "I don't mind" than "if I have to"—less of a negative connotation I think!

I apologise for the slower response, it's been a bank holiday weekend here and I'm a little out of whack!

**Reply by tim-schilling** [reply]
The meeting invites have been sent out for June 7th. I look forward to talking with folks then!

**Reply by tim-schilling** [reply]
Following up here, on the responsibilities tab of the sheet on the event invite has longer form descriptions or a linked playbook for the tasks. Please let me know where I'm not providing enough information or context so I can improve it.

**Comment by ryancheley** [comment]
@tim-schilling I somehow missed the google form and the calendly. Can you send me the invite as well. Looking at the times from Calendly for June 7th I should be available. Sorry about this!

**Reply by carltongibson** [reply]
June 7th is right during DjangoCon Europe. Could we do it the week after instead?

**Reply by tim-schilling** [reply]
It's also during a wagtail space event. I'll cancel those and start the poll over again.

**Comment by joshuadavidthomas** [comment]
@tim-schilling Just getting around to answering the survey and the Calendly poll. Apologies for the lateness, I've been out-of-pocket since last Friday morning.

**Comment by tim-schilling** [comment]
Alright, let's try this again. Meeting poll for June 17 to June 26th. https://xoyondo.com/dp/wa1thb94i9xw1ju

**Comment by tim-schilling** [comment]
On a related note, there's a package maintainer working group proposal here: https://github.com/django/dsf-working-groups/pull/19

**Comment by cunla** [comment]
In terms of managing the github org/teams access - I have done it in the past for a company and created a tool to do it. The tool I created is proprietary, but I found something similar here https://github.com/cilium/team-manager.
I am happy to take over this - anyway, I thought about creating an open-source tool to do it :smile:

Would it be possible to add to the agenda:
- What packages is this group planning to maintain? i.e., what criteria for accepting new packages?
- What packages are already maintained by the group?

**Reply by tim-schilling** [reply]
Let me take a stab at answering these questions ahead of time.

> What packages is this group planning to maintain?

The admins won't be maintaining anything. They are facilitating the community to maintain packages. If you're asking what packages would qualify, any package that could be used within a Django project (IMO). And then see below for repo requirements.

> What criteria for accepting new packages?

I have a initial set of requirements for a repo here: https://github.com/django-commons/membership/blob/main/incoming_repo_requirements.md

> What packages are already maintained by the group?

None. I don't want to transfer any projects until I have at least one other person as an admin. @matthiask and I would like to [move the debug toolbar](https://github.com/jazzband/django-debug-toolbar/issues/1918) and there's a [request to move django-fsm-2](https://github.com/django-commons/membership/issues/25) already open.

Regarding managing teams access, I think having a configuration file would be a better approach than what I have currently. It would be easier to add/remove and for people to request changing teams. Open a issue a new issue so we can vet that process later.

**Reply by cunla** [reply]
Sorry, I am not sure I understood the last sentence - should I open an issue to check the org management using a config file process?

**Reply by tim-schilling** [reply]
Sorry about that! Please open an issue to discuss moving managing membership via config file with either [cilium/team-manager](https://github.com/cilium/team-manager.), https://github.com/eclipse-csi/otterdog or something similar.

**Comment by tim-schilling** [comment]
Hi folks, thank you for attending the calls today. I hope I was able to provide enough information to folks who were unclear before. 

As we talked about, the next steps are:
1. Update the ["Revised responses" tab](https://docs.google.com/spreadsheets/d/1UThBMs0lTs6mcIRAY8q2YrNzASRnVrF0ojOiYkp9Bws/edit?usp=sharing) with which you'd be interested in
2. Next week around this time, I'll aggregate the responses and fill out our teams (conduct, membership, PyPI admins, and security)
3. I'll estimate the effort for each team's time
4. I'll reach out to folks to confirm they are okay with the responsibilities and time commitment for the next ~6 months
5. We'll onboard people, sand down the rough edges
6. We'll accept django-fsm-2 (https://github.com/django-commons/membership/issues/25), django-captcha and start the process to move Django Debug Toolbar

The notes I have from our discussions are:
- Begin with an eye on capacity, need to avoid bringing projects on without maintainers
- Create a plan for handling a project that becomes unmaintained (README banner, etc.)
- The initial set of responsibilities seems like a lot
- Someone asked, "Will organizers need to take over maintenance?" - The answer is no
- We need a place for people in the community to communicate (discussions on the org, section in the forum and #packages on Django discord)
- Regarding inbound project requirements, we need to be cautious about having too many barriers to entry
- Some folks are interested in fostering the community in regard to package maintenance and helping everyone improve

**Reply by marksweb** [reply]
Turns out those status [badges](https://shields.io/badges) are way easier than I realised.

https://img.shields.io/badge/django_commons-status-blue

**Reply by benbacardi** [reply]
Apologies I couldn't make the meeting, it looks like it was a productive one! Thanks for the notes, Tim. Let me know how I can help going forwards.

**Comment by nanorepublica** [comment]
Argh, I don't (yet) notifications for new discussions, I would have been up for this completely had I seen it 🤦 

Obviously I wasn't at the meeting, but something that might aid inbound projects is to have a template repo that has the expected requirements done to a certain extent? Mainly github workflows/actions etc.

**Comment by tim-schilling** [comment]
Hi folks, I'm circling back around on this. I invited @williln, @Stormheg, @ryancheley and @cunla to be admins for the org and they've all accepted. The reason I chose them over others was that they were actively interested in doing the admin tasks within github and the check-ins amongst the membership. I feel like that level of logistics and desire to work within the community is the biggest priority for Django Commons at this time.

However, the five of us don't cover all of the needs very well. Specifically around the Security and CoC teams. We're hoping that those teams will look like the following.

Security team
- Lacey
- Storm
- @carltongibson 
- @marksweb 

CoC team
- Tim
- @carltongibson 
- @matthiask 
- @knyghty

I will follow-up with each of you individually on your admin request issues to confirm if that's alright with you.

If anyone has any questions or concerns, please feel free to ask here or email me directly. I appreciate everyone for taking the time to participate in my questionnaire on the calls!
